projects/plugins/boost/app/assets/src/js/utils/cloud-css.ts

```diff
@@ -2,7 +2,43 @@
  * Internal dependencies
  */
 import api from '../api/api';
+import { getStatus, updateStatus, resetStatus, CloudCssStatus } from '../stores/cloud-css-status';
 
-export function requestCloudCss(): Promise< string > {
-	return api.post( '/cloud-css/request-generate' );
+export async function requestCloudCss(): Promise< void > {
+	// Todo: Debounce request.
+	resetStatus();
+	await api.post( '/cloud-css/request-generate' );
+	pollCloudCssStatus();
+}
+
+let statusIntervalId = null;
+
+/**
+ * Poll Cloud Critical CSS on regular intervals.
+ *
+ * If there are no active request made to generate Cloud CSS from the client,
+ * poll the status with a long interval.
+ *
+ * If there is an active request made to generate Cloud CSS from the client,
+ * poll the status with a short interval. Once there are no pending responses,
+ * poll the status with a long interval.
+ */
+export function pollCloudCssStatus() {
+	let status = getStatus();
+	const shortInterval = 5000;
+	const longInterval = 2 * 60 * 1000;
+	const interval = status.pending ? shortInterval : longInterval;
+
+	if ( statusIntervalId !== null ) {
+		clearInterval( statusIntervalId );
+	}
+
+	statusIntervalId = setInterval( async () => {
+		status = await api.get< CloudCssStatus >( '/cloud-css/status' );
+		updateStatus( status );
+
+		if ( ! status.pending && interval === shortInterval ) {
```

Comment:
This is much better. Though one final improvement suggestion, just to make the code super readable and easy to follow:

Add a function to this module called `pollIntervalForStatus( status )` which accepts a `status` object.

Then the start of this function can say:
```
const interval = pollIntervalForStatus( getStatus() );
```

And whenever you check if the interval should change, you can change this `if` to:
```
if ( interval !== pollIntervalForStatus( status ) )
```

That will cover both cases -- any time the interval wants to change because the status has changed, it will re-call `pollCloudCssStatus`.

What do you think? Just a nice change to make the code super easy to read.

Reply:
Oh. Why didn't I think that! It works like a charm. Thanks @thingalon